src/NHibernate/Cache/ReadWriteCache.cs

```diff
@@ -241,7 +220,7 @@ public ISoftLock Lock(CacheKey key, object version)
 				{
 					log.Debug("Caching: {0}", string.Join(",", keys.AsEnumerable()));
 				}
-				var keysArr = keys.Cast<object>().ToArray();
+				var keysArr = keys.ToArray<object>();
```

Comment:
Why this conversion is required? You can call `_cache.LockMany(keys);` directly

Reply:
I don't know, I just removed `Cast` which is not needed.

Reply:
I think this was done to prevent co-variant array conversion complaint by R#

Reply:
This array is not expected to be modified so it shouldn't be an issue. IMHO we should remove those conversions if R# complaint is the only reason.

Reply:
Done